presto-main/src/main/java/com/facebook/presto/operator/UnnestOperator.java

```diff
@@ -112,7 +113,12 @@ public UnnestOperator(OperatorContext operatorContext, List<Integer> replicateCh
         this.unnesters = new ArrayList<>(unnestTypes.size());
         for (Type type : unnestTypes) {
             if (type instanceof ArrayType) {
-                unnesters.add(new ArrayUnnester(((ArrayType) type).getElementType()));
+                if (((ArrayType) type).getElementType() instanceof RowType) {
```

Comment:
Might consider put a local variable `elementType = ((ArrayType) type).getElementType()`  and being reused ?